pkg/status/handlers.go

```diff
@@ -27,7 +34,7 @@ type API struct {
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/status", a.alive)
 	mux.Get("/api/v0/version", a.version)
-
+	mux.Get("/api/v0/deepcheck", a.deepCheck)
```

Comment:
I have two general questions for this new API endpoint:

- Who will be the consumer/caller of this API endpoint?
- I understand that we want to check the connectivities between services. Using this method means we need to change this endpoint implementation whenever we want to check more upstream dependencies. I think it would be better if we do it in another way. We could use the circuit breaker middleware responsible for managing the states of breakers for any upstream dependency services. It can then provide us with more metrics of the breakers' states, which alternatively show any issues that happen between login ui and upstream dependency, or upstream dependency itself.

Reply:
For the first point: I'd like a pebble check to call this route
For the second point: Interesting idea. Two points: I haven't implemented circuit breakers yet, but would love to learn. Before moving in this direction, I want to get @shipperizer  and @nsklikas thought's on it. We want to implement the deep health check to get prometheus metrics about the dependencies' health. For now I have a binary one/zero metric to suggest if a component is up, or down.

Reply:
@wood-push-melon idea is to have a user to call this and have metrics around it at each call

next step would be to have the 2 checks running on the background and update a object that the endpoint is simply exposing it

the idea of the middleware is a good one, but i need to see it and understand it properly before being able to tell if it makes sense for this specific case, as we mainly want something really simple, not a fully fledged "status" page like [github.status](https://www.githubstatus.com/)